js/modules/loki_file_server_api.js

```diff
@@ -43,6 +53,182 @@ class LokiFileServerAPI {
     return this._setOurDeviceMapping(authorisations, isPrimary);
   }
 
+  async getDeviceMappingForUsers(pubKeys) {
+    const users = await this._server.getUsers(pubKeys);
+    return users;
+  }
+
+  async verifyUserObjectDeviceMap(pubKeys, isRequest, iterator) {
+    const users = await this.getDeviceMappingForUsers(pubKeys);
+    // log.info('verifyUserObjectDeviceMap Found', users.length, 'users')
+
+    // go through each user and find deviceMap annotations
+    const notFoundUsers = [];
+    users.forEach(user => {
+      let found = false;
+      if (!user.annotations || !user.annotations.length) {
+        log.info(
+          `verifyUserObjectDeviceMap no annotation for ${user.username}`
+        );
+        return;
+      }
+      user.annotations.forEach(note => {
+        if (note.type !== 'network.loki.messenger.devicemapping') {
```

Comment:
I think using a loop here sends the wrong message.
What we really want is handle the one annotation that has a type of `DEVICE_MAPPING_ANNOTATION_KEY` .

Something like

```js
const note = user.annotations.find(x => x.type === DEVICE_MAPPING_ANNOTATION_KEY);
if (!note) {
    // device mapping not present, log if necessary
    return;
}
// do stuff with note